### PR TITLE
Branch1

### PR DESCRIPTION
**Pull request by HassanLotfyAbdelhamid**
Pull request 1\ RCF file generator

### CODE REVIEW

File: XML to RCF Editor2/new_file.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<parent active="yes">
```
Comment (Merna-appraider) [inline XML to RCF Editor2/new_file.xml]: please remove this file

File: XML to RCF Editor2/new_file.rcf [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<parent active="yes">
```
Comment (Merna-appraider) [inline XML to RCF Editor2/new_file.rcf]: please consider active =yes on the root rule please, ex. Rule-11.1

File: XML to RCF Editor2/my_xml.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<parent>
```
Comment (Merna-appraider) [inline XML to RCF Editor2/my_xml.xml:1]: please make any mapped="yes" or active="yes" as no

File: XML to RCF Editor2/functions.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+def parse_xml_return_root(file_name_with_path):
+
+    # error handling in case directory in incorrect or file content syntax is invalid
+    try:
+        # parse the xml file
+        tree = ET.parse(rf"{file_name_with_path}")
+        root = tree.getroot()
+        return root
+    except FileNotFoundError:
```
Comment (Merna-appraider) [inline XML to RCF Editor2/functions.py]: please remove this exception

File: XML to RCF Editor2/functions.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    # error handling in case directory in incorrect or file content syntax is invalid
+    try:
+        # parse the json file
+        with open(file_name_with_path, 'r') as file:
+            json_dictionary = json.load(file)
+            return json_dictionary
+    except FileNotFoundError:
+        print("File not found")
+        return None
+    except json.JSONDecodeError:
```
Comment (Merna-appraider) [inline XML to RCF Editor2/functions.py]: please add this exception for line 41

File: XML to RCF Editor2/functions.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+                new_message.set(message_key, msg)
+                new_message.set(key_to_change, key_to_change_val)
+                # append the clone to the XML file
+                enforcement.append(new_message)
+                print(f"message with id = {msg} under rule id {element.get("id")} does not exist. message added")
+
+
+
+"""
+function name: find_and_edit
```
Comment (Merna-appraider) [inline XML to RCF Editor2/functions.py]: why are we having this function?

File: XML to RCF Editor2/functions.py [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+                new_message.set(req_att_name, req_att_val)
+                # append the clone to the XML file
+                parent2.append(new_message)
+                print("message does not exist. message added")
+
+            return True
+
+
+"""
+function name: update_into_rcf
```
Comment (Merna-appraider) [inline XML to RCF Editor2/functions.py:148]: please remove the handling of xml saving

File: XML to RCF Editor2/app.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
+
+import xml.etree.ElementTree as ET
+
+import functions
+import argparse
+
+import json
+
+parser = argparse.ArgumentParser(description="Script to edit xml files and save it as .rcf")
```
Comment (Merna-appraider) [inline XML to RCF Editor2/app.py]: please put in function called handle_arguments

File: XML to RCF Editor2/app.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+# parse the json file into dictionary
+json_dictionary = functions.parse_json_return_dict(args.input_json_file_name)
+
+
+########## modifying ###########
+if json_dictionary is not None and root is not None:
+    functions.loop_and_enable(root, json_dictionary)
+
+# # search for message id= '0302' and edit its value
+# functions.find_and_edit(root,'id','Rule-11.1',
```
Comment (Merna-appraider) [inline XML to RCF Editor2/app.py]: please remove any commented code

**Review by Merna-appraider — CHANGES_REQUESTED** [review]

File: XML to RCF Editor2/functions.py (hunk 6 above)
Comment (HassanLotfyAbdelhamid) [reply XML to RCF Editor2/functions.py]: this function was for testing. function removed from functions.py and added to test.py

**Pull request closed without merging**